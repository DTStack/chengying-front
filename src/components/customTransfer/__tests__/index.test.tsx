import * as React from 'react';
import CustomTransfer from '../index';
import {
  fireEvent,
  render,
  RenderResult,
  cleanup,
} from '@testing-library/react';

const defaultProps = {
  dataSource: [
    { id: '172.16.8.59', ip: '172.16.8.59', key: '172.16.8.59' },
    { id: '172.16.8.76', ip: '172.16.8.76', key: '172.16.8.76' },
    { id: '172.16.8.57', ip: '172.16.8.57', key: '172.16.8.57' },
  ],
  filterOption: () => null,
  footerLeft: (props) => null,
  footerRight: (props) => null,
  leftPageCurrent: 1, // 左边当前页码
  rightPageCurrent: 1, // 右边当前页码
  realTargetKeys: ['172.16.8.57'], // 全部已经选择的keys
  onChange: jest.fn(),
  listStyle: {},
  selectedKeys: [],
  showSearch: true,
  rowKey: () => null,
  targetKeys: ['172.16.8.57'],
  totalData: [
    { id: 2, ip: '172.16.8.59' },
    { id: 3, ip: '172.16.8.76' },
  ], // 所有数据（已选+未选）
  isRightPanelCheckoutAll: false,
  render: () => null,
};

describe('CustomTransfer unit test', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = render(<CustomTransfer {...defaultProps} />);
  });

  afterEach(cleanup);

  test('CustomTransfer table render', () => {
    const left = wrapper.getByTestId('left');
    expect(left).not.toBe('10 / page');
    const tr = left.getElementsByTagName('tr');
    expect(tr.length).toBe(3);
    for (const i in tr) {
      if (tr[i] instanceof HTMLElement) {
        expect(tr[i].getAttribute('data-row-key')).toBe(
          defaultProps.dataSource[i].ip
        );
      }
    }
  });

  // rerender
  test('CustomTransfer snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
