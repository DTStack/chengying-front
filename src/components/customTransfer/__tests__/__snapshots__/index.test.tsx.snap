// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTransfer unit test CustomTransfer snapshot 1`] = `
<DocumentFragment>
  <div
    class="transfer-table ant-transfer ant-transfer-customize-list"
  >
    <div
      class="ant-transfer-list"
    >
      <div
        class="ant-transfer-list-header"
      >
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-transfer-list-header-selected"
        >
          <span>
            3 items
          </span>
          <span
            class="ant-transfer-list-header-title"
          />
        </span>
      </div>
      <div
        class="ant-transfer-list-body ant-transfer-list-body-with-search"
      >
        <div
          class="ant-transfer-list-body-search-wrapper"
        >
          <div>
            <input
              class="ant-input ant-transfer-list-search"
              placeholder="Search here"
              type="text"
              value=""
            />
            <span
              class="ant-transfer-list-search-action"
            >
              <i
                aria-label="icon: search"
                class="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
        <div
          class="ant-transfer-list-body-customize-wrapper"
        >
          <div
            class="ant-table-wrapper dt-table-last-row-noborder dt-table-fixed-contain-footer"
            style="height: 313px;"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-small ant-table-without-column-header ant-table-fixed-header ant-table-scroll-position-left ant-table-layout-fixed"
                  data-testid="left"
                >
                  <div
                    class="ant-table-content"
                  >
                    <div
                      class="ant-table-scroll"
                    >
                      <div
                        class="ant-table-body"
                        style="overflow-y: scroll;"
                        tabindex="-1"
                      >
                        <table
                          class=""
                        >
                          <colgroup>
                            <col
                              class="ant-table-selection-col"
                            />
                            <col />
                          </colgroup>
                          <tbody
                            class="ant-table-tbody"
                          >
                            <tr
                              class="ant-table-row ant-table-row-level-0"
                              data-row-key="172.16.8.59"
                            >
                              <td
                                class="ant-table-selection-column"
                              >
                                <span>
                                  <label
                                    class="ant-checkbox-wrapper"
                                  >
                                    <span
                                      class="ant-checkbox"
                                    >
                                      <input
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </span>
                              </td>
                              <td
                                class=""
                              />
                            </tr>
                            <tr
                              class="ant-table-row ant-table-row-level-0"
                              data-row-key="172.16.8.76"
                            >
                              <td
                                class="ant-table-selection-column"
                              >
                                <span>
                                  <label
                                    class="ant-checkbox-wrapper"
                                  >
                                    <span
                                      class="ant-checkbox"
                                    >
                                      <input
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </span>
                              </td>
                              <td
                                class=""
                              />
                            </tr>
                            <tr
                              class="ant-table-row ant-table-row-level-0"
                              data-row-key="172.16.8.57"
                            >
                              <td
                                class="ant-table-selection-column"
                              >
                                <span>
                                  <label
                                    class="ant-checkbox-wrapper"
                                  >
                                    <span
                                      class="ant-checkbox"
                                    >
                                      <input
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </span>
                              </td>
                              <td
                                class=""
                              />
                            </tr>
                          </tbody>
                        </table>
                      </div>
                    </div>
                  </div>
                </div>
                <ul
                  class="ant-pagination ant-table-pagination mini"
                  unselectable="unselectable"
                >
                  <li
                    aria-disabled="true"
                    class="ant-pagination-disabled ant-pagination-prev"
                    title="Previous Page"
                  >
                    <a
                      class="ant-pagination-item-link"
                    >
                      <i
                        aria-label="icon: left"
                        class="anticon anticon-left"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="left"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                          />
                        </svg>
                      </i>
                    </a>
                  </li>
                  <li
                    class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                    tabindex="0"
                    title="1"
                  >
                    <a>
                      1
                    </a>
                  </li>
                  <li
                    aria-disabled="true"
                    class="ant-pagination-disabled ant-pagination-next"
                    title="Next Page"
                  >
                    <a
                      class="ant-pagination-item-link"
                    >
                      <i
                        aria-label="icon: right"
                        class="anticon anticon-right"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                          />
                        </svg>
                      </i>
                    </a>
                  </li>
                  <li
                    class="ant-pagination-options"
                  >
                    <div
                      class="ant-select-sm ant-pagination-options-size-changer ant-select ant-select-enabled"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-controls="test-uuid"
                        aria-expanded="false"
                        aria-haspopup="true"
                        class="ant-select-selection
            ant-select-selection--single"
                        role="combobox"
                        tabindex="0"
                      >
                        <div
                          class="ant-select-selection__rendered"
                        >
                          <div
                            class="ant-select-selection-selected-value"
                            style="display: block; opacity: 1;"
                            title="10 / page"
                          >
                            10 / page
                          </div>
                        </div>
                        <span
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <i
                            aria-label="icon: down"
                            class="anticon anticon-down ant-select-arrow-icon"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </i>
                        </span>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-transfer-operation"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
        disabled=""
        type="button"
      >
        <i
          aria-label="icon: right"
          class="anticon anticon-right"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </i>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
        disabled=""
        type="button"
      >
        <i
          aria-label="icon: left"
          class="anticon anticon-left"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </i>
      </button>
    </div>
    <div
      class="ant-transfer-list"
    >
      <div
        class="ant-transfer-list-header"
      >
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-transfer-list-header-selected"
        >
          <span>
            0 item
          </span>
          <span
            class="ant-transfer-list-header-title"
          />
        </span>
      </div>
      <div
        class="ant-transfer-list-body ant-transfer-list-body-with-search"
      >
        <div
          class="ant-transfer-list-body-search-wrapper"
        >
          <div>
            <input
              class="ant-input ant-transfer-list-search"
              placeholder="Search here"
              type="text"
              value=""
            />
            <span
              class="ant-transfer-list-search-action"
            >
              <i
                aria-label="icon: search"
                class="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
        <div
          class="ant-transfer-list-body-customize-wrapper"
        >
          <div
            class="ant-table-wrapper dt-table-last-row-noborder dt-table-fixed-contain-footer"
            style="height: 313px;"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-small ant-table-empty ant-table-without-column-header ant-table-fixed-header ant-table-scroll-position-left ant-table-layout-fixed"
                  data-testid="right"
                >
                  <div
                    class="ant-table-content"
                  >
                    <div
                      class="ant-table-scroll"
                    >
                      <div
                        class="ant-table-body"
                        style="overflow-y: scroll;"
                        tabindex="-1"
                      >
                        <table
                          class=""
                        >
                          <colgroup>
                            <col
                              class="ant-table-selection-col"
                            />
                            <col />
                          </colgroup>
                          <tbody
                            class="ant-table-tbody"
                          />
                        </table>
                      </div>
                      <div
                        class="ant-table-placeholder"
                      >
                        <div
                          class="ant-empty ant-empty-normal"
                        >
                          <div
                            class="ant-empty-image"
                          >
                            <svg
                              height="41"
                              viewBox="0 0 64 41"
                              width="64"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                                transform="translate(0 1)"
                              >
                                <ellipse
                                  cx="32"
                                  cy="33"
                                  fill="#F5F5F5"
                                  rx="32"
                                  ry="7"
                                />
                                <g
                                  fill-rule="nonzero"
                                  stroke="#D9D9D9"
                                >
                                  <path
                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                  />
                                  <path
                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    fill="#FAFAFA"
                                  />
                                </g>
                              </g>
                            </svg>
                          </div>
                          <p
                            class="ant-empty-description"
                          >
                            No Data
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
